kfp_notebook/tests/test_notebook_op.py

```diff
@@ -0,0 +1,209 @@
+#
+# Copyright 2018-2020 IBM Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import pytest
+from kfp_notebook.pipeline import NotebookOp
+
+
+@pytest.fixture
+def notebook_op():
+    return NotebookOp(name="test",
+                      notebook="test_notebook.ipynb",
+                      cos_endpoint="http://testserver:32525",
+                      cos_bucket="test_bucket",
+                      cos_directory="test_directory",
+                      cos_dependencies_archive="test_archive.tgz",
+                      image="test/image:dev")
+
+
+def test_fail_without_cos_endpoint():
+    with pytest.raises(TypeError) as error_info:
+        NotebookOp(name="test",
+                   notebook="test_notebook.ipynb",
+                   cos_bucket="test_bucket",
+                   cos_directory="test_directory",
+                   cos_dependencies_archive="test_archive.tgz",
+                   image="test/image:dev")
```

Comment:
Is the fieldname listed in the exception?  If so, you might want to confirm that the appropriate field is in the message.

Reply:
confirmed fieldname is listed in the error (TypeError)
TypeError: __init__() missing 1 required positional argument: 'cos_endpoint'